### PR TITLE
Authentication configs are fixed [API-1080]

### PR DESCRIPTION
**Pull request by hakanaktas0**
Fix for [this issue](https://github.com/hazelcast/hazelcast-cpp-client/issues/944)

### CODE REVIEW

File: Reference_Manual.md [hunk 1]
```diff
@@ -1269,22 +1271,74 @@ The following is an example configuration where we set the secret token bytes to
...
-                    <token encoding="base64">SGF6ZW</token>
-                </identity>
-            </realm>
-        </realms>
-    </security>
+    <realms>
+        <realm name="jaasRealm">
+            <authentication>
+                <jaas>
+                    <login-module class-name="path.to.file.TokenLoginModule" usage="REQUIRED">
```
Comment (ihsandemir) [inline Reference_Manual.md:1331]: is this path or package name?

Comment (hakanaktas0) [reply Reference_Manual.md:1331]: path

File: Reference_Manual.md [hunk 2]
```diff
@@ -1269,22 +1271,74 @@ The following is an example configuration where we set the secret token bytes to
...
+                    </login-module>
+                </jaas>
+            </authentication>
+            <identity>
+                <token>SGF6ZW</token>
+            </identity>
+        </realm>
+    </realms>
+    <client-authentication realm="jaasRealm"/>
```
[the remark this answers is not in the document]
Comment (hakanaktas0) [reply Reference_Manual.md:1340]: I am surprised too, but the above one doesn't work with `<client-authentication realm=""/>` for some reason, client just connects whether the credentials are correct or not, when configured that way. I tried them both for both cases and this is how they work.

**Comment by hakanaktas0** [comment]
Replaced with [this](https://github.com/hazelcast/hazelcast-cpp-client/pull/1011)

**Pull request closed without merging**